platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/FilterInjector.java

```diff
@@ -0,0 +1,202 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import javax.servlet.FilterRegistration;
+import javax.servlet.ServletContext;
+import javax.servlet.SessionCookieConfig;
+import javax.servlet.http.HttpSessionListener;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Injects the {@link SecurityJavaSubjectFilter} into the {@link ServletContext} with properties
+ * filterName = {@link FilterInjector#SECURITY_JAVA_SUBJECT_FILTER}, urlPatterns = {@link
+ * FilterInjector#ALL_URLS}, and no servlet name as the {@link ServletContext} is being created.
+ * Sets up all {@link javax.servlet.Servlet}s with the {@link SecurityJavaSubjectFilter} in the
+ * {@link javax.servlet.FilterChain}.
+ */
+public class FilterInjector implements EventListenerHook {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FilterInjector.class);
+
+  private static final String ALL_URLS = "/*";
+
+  private static final String SECURITY_JAVA_SUBJECT_FILTER = "security-java-subject-filter";
+
+  private final Filter filter;
+
+  private final List<HttpSessionListener> sessionListeners = new ArrayList<>();
+
+  private final ScheduledExecutorService executorService;
+
+  /**
+   * Creates a new filter injector with the specified filter.
+   *
+   * @param filter filter that should be injected.
+   * @param executorService used to check for missed servlet contexts
+   */
+  public FilterInjector(Filter filter, ScheduledExecutorService executorService) {
+    this.filter = filter;
+    this.executorService = executorService;
+  }
+
+  @Nullable
+  BundleContext getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(SecurityJavaSubjectFilter.class);
+    if (cxfBundle != null) {
+      return cxfBundle.getBundleContext();
+    }
+    return null;
+  }
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectFilter((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  public void init() {
+    executorService.schedule(this::checkForMissedServletContexts, 1, TimeUnit.SECONDS);
+  }
+
+  public void destroy() {
```

Comment:
Does this destroy method work?  I think it needs an `int code` argument.
https://github.com/codice/ddf/blob/master/catalog/transformer/catalog-transformer-html/src/main/java/ddf/catalog/transformer/html/models/HtmlCategoryModel.java#L48
https://issues.apache.org/jira/browse/ARIES-1436

Reply:
It get's called correctly. Maybe it doesn't need since it's listed in the blueprint as the `destroy-method`.

Reply:
If it's blueprint then I think it's needed.

Reply:
@bdeining adding `int code` caused bundle to fail to resolve resulting in itests to fail.
`Component 'filterInjector' does not have destroy-method: destroy`

I see that your example is using blueprint the same way so I can't really say way this is...